Ensure user email is always in email address table

ivanvc opened the pull request:
This PR solves the issue with duplicated emails reported in #11251, by ensuring that the primary user email address is in the email_address table too.

Fixes #11251.

lafriks commented:
I don't think this can be backported anymore as there is migration

6543 commented:
> I don't think this can be backported anymore as there is migration

this should go into **doctor** as task

ivanvc commented:
Finally, I was able to fix the fixtures that were causing errors in some unit + integration tests.

It does need the migration, else users may be able to duplicate their email addresses.

a1012112796 commented on models/fixtures/user.yml:
```diff
@@ -186,14 +186,14 @@
   name: user11
   full_name: User Eleven
-  email: user11@example.com
+  email: user011@example.com
```
why change ?

ivanvc replied:
Because in the `email_1ddress.yml` fixtures, the `user11@example.com` is registered to `user1`, and changing that file rather than this, brings more issues and places where it should be updated (I initially went through that route with commits bdd4bc2d827fce00328493823fe3bd9e0359d73a and 77cd65708e5e56a11ae9017123a77ae35f4fe7fb), and decided later that the minimal impact would be to update the email address here.

a1012112796 commented on models/migrations/v158.go:
```diff
@@ -0,0 +1,80 @@
+		IsActivated bool
+	}
+
+	if err := x.Sync2(new(User), new(EmailAddress)); err != nil {
+		return err
+	}
```
I don't think it's necessary, because you haven't changeed the struct of them.

ivanvc replied:
Got it, my bad, just read some other migrations and I thought it was required.

a1012112796 commented:
In my opion, maybe it's not necessary to kept ``Email`` cols in the ``user``, get it from EmailAddress table when needed is enough.

ivanvc commented:
> n my opion, maybe it's not necessary to kept `Email` cols in the `user`, get it from EmailAddress table when needed is enough.

That was my original intention, as I removed the Email field from the User, havoc broke, as it is used in so many places that I reverted to better sync the fields in the user and emails tables.

This PR has the minimum to fix the issue, if a completely re-write of the User model is what is intended (or best), I think this should be closed then, as it will be a way bigger effort.

lunny commented:
Please resolve the conflicts.

ivanvc commented:
I'm not sure if actually solving the conflicts makes sense anymore. Probably, as it is very stale, it makes more sense to actually re-work it. I'm gonna close the PR and will open a new one later, whenever I get some free time.
